src/operator/contrib/proposal.cu

```diff
@@ -552,8 +557,8 @@ class ProposalGPUOp : public Operator{
     // copy nms result to gpu
     int* keep;
     FRCNN_CUDA_CHECK(cudaMalloc(&keep, sizeof(int) * _keep.size()));
-    FRCNN_CUDA_CHECK(cudaMemcpy(keep, &_keep[0], sizeof(int) * _keep.size(),
-                                cudaMemcpyHostToDevice));
+    FRCNN_CUDA_CHECK(cudaMemcpyAsync(keep, &_keep[0], sizeof(int) * _keep.size(),
+                                     cudaMemcpyHostToDevice, stream));
```

Comment:
Here, memcpy is being scheduled to run on `stream`, but [line 566](https://github.com/apache/incubator-mxnet/blob/743c10c3d931b84de71a3731a69c6494921dd43d/src/operator/contrib/proposal.cu#L566) runs `PrepareOutput` on the default context. Unless I am missing something, this introduces a potential race condition: when `stream` does not match the default stream, `PrepareOutput` may run before `cudaMemcpyAsync` completes.

Reply:
Same.